tests/bgp/test_bgpmon.py

```diff
@@ -102,14 +125,22 @@ def build_syn_pkt(local_addr, peer_addr):
     exp_packet.set_ignore_extra_bytes()
     return exp_packet
 
-def test_bgpmon(duthost, localhost, common_setup_teardown, ptfadapter, ptfhost):
+
+def test_bgpmon(dut_with_default_route, localhost, common_setup_teardown, ptfadapter, ptfhost):
     """
     Add a bgp monitor on ptf and verify that DUT is attempting to establish connection to it
     """
+
+    duthost = dut_with_default_route
+
     def bgpmon_peer_connected(duthost, bgpmon_peer):
         try:
-            bgp_summary = json.loads(duthost.shell('vtysh -c "show bgp summary json"')['stdout'])
-            return bgp_summary['ipv4Unicast']['peers'][bgpmon_peer]["state"] == "Established"
+            bgp_summary_all_asics = duthost.run_vtysh('-c "show bgp summary json"', asic_index='all')
```

Comment:
line 138 works because duthost is a MultiAsicSonicHost which has __getattr___ overwritten at https://github.com/sonic-net/sonic-mgmt/blob/master/tests/common/devices/multi_asic.py#L252 to work on the specified asics. 

We will try out using enum_rand_one_frontend_asic_index and get back to you

Reply:
thanks @mannytaheri , bgp tests can check per asic to save much more time